### PR TITLE
Feature/deactivate account fix

### PR DESCRIPTION
**Pull request by Filienko**
Creating CLI commands to disable and enable active as per https://www.pivotaltracker.com/story/show/185455581

### CODE REVIEW

File: isacc_messaging/api/isacc_record_creator.py [hunk 1]
```diff
@@ -364,7 +364,7 @@ def on_twilio_message_received(self, values):
         })
         pt = first_in_bundle(pt)
         if not pt:
-            error = "No patient with this phone number"
+            error = "No patient with this phone number"
```
Comment (pbugni) [inline isacc_messaging/api/isacc_record_creator.py]: trailing whitespace?

File: isacc_messaging/models/isacc_patient.py [hunk 2]
```diff
@@ -32,6 +32,28 @@ def __repr__(self):
...
+        response = HAPI_request('GET', 'Patient', params={
+            "active": "true"
+        })
+        return response
+
+    @staticmethod
+    def get_patient_by_id(id):
+        """Execute query for active patients
+
+        NB, returns only patients with active set to true
```
Comment (pbugni) [inline isacc_messaging/models/isacc_patient.py]: i see no check for active here, as the comment suggests?

File: isacc_messaging/models/isacc_patient.py [hunk 3]
```diff
@@ -32,6 +32,28 @@ def __repr__(self):
...
+        """
+        response = HAPI_request('GET', 'Patient', params={
+            "id": id
+        })
+        return response
+    
+    @staticmethod
+    def all_patients():
+        """Execute query for all patients
+
```
Comment (pbugni) [inline isacc_messaging/models/isacc_patient.py:45]: again, i don't see any code checking for active patients.

**Review by pbugni — CHANGES_REQUESTED** [review]

File: isacc_messaging/api/views.py [hunk 4]
```diff
@@ -202,3 +202,39 @@ def update_patient_extensions(dry_run):
...
+
+
+@base_blueprint.cli.command("deactivate_patient")
+@click.argument('patient_id')
+def deactivate_patient(patient_id):
+    """Iterate through all patients, update the active parameter value based on id"""
+    from isacc_messaging.models.fhir import next_in_bundle
+    from isacc_messaging.models.isacc_patient import IsaccPatient as Patient
+    active_patients = Patient.all_patients()
+    for json_patient in next_in_bundle(active_patients):
```
Comment (pbugni) [inline isacc_messaging/api/views.py]: why not just search for the single patient by given id?

File: isacc_messaging/api/views.py [hunk 5]
```diff
@@ -202,3 +202,39 @@ def update_patient_extensions(dry_run):
...
+    """Iterate through all patients, activate all of them"""
+    from isacc_messaging.models.fhir import next_in_bundle
+    from isacc_messaging.models.isacc_patient import IsaccPatient as Patient
+    all_patients = Patient.all_patients()
+    for json_patient in next_in_bundle(all_patients):
+        patient = Patient(json_patient)
+        patient.active = True
+        patient.persist()
+        audit_entry(
+        f"Patient with id {patient.id} reinstated",
```
Comment (pbugni) [inline isacc_messaging/api/views.py]: audit's are intended for paper trail.  i'd expect this CLI is typically only going to be used once - when this feature is turned on.  i'd suggest "Patient {patient.id} active set true" or the like.

**Review by pbugni — CHANGES_REQUESTED** [review]

**Review by pbugni — CHANGES_REQUESTED** [review]
This looks good, but I'm concerned about `on_twilio_message_received()`

We don't include the `active` check when looking up a patient there, so if there happens to be 2 patients with the same phone number, `first_in_bundle()` could give you the in-active one.

I think you should modify the query there, and remove the `pt.active` checks below there.

File: isacc_messaging/api/isacc_record_creator.py [hunk 6]
```diff
@@ -236,8 +236,8 @@ def generate_incoming_message(self, message, time: datetime = None, patient: Pat
         if priority is None:
             priority = "routine"
 
-        if patient is None:
-            return "Need patient"
+        if patient is None or patient.active != True:
+            return "Need active patient"
```
Comment (pbugni) [inline isacc_messaging/api/isacc_record_creator.py]: i realize this is just a change from before, but this is close to an unnecessary check.  if we're going to check for a required parameter and it isn't there or valid, please raise an exception.

**Review by pbugni — CHANGES_REQUESTED** [review]
one minor nit to fix.

File: isacc_messaging/models/isacc_patient.py [hunk 7]
```diff
@@ -32,12 +32,31 @@ def __repr__(self):
...
+        return response
+
+    @staticmethod
+    def all_patients():
+        """Execute query for all patients
+
         NB, until status is set on all patients, queries for
         any status/active value will skip those without a value.
         """
         response = HAPI_request('GET', 'Patient')
```
Comment (achen2401) [inline isacc_messaging/models/isacc_patient.py:49]: @Filienko Do you know if this does indeed return **all** the patients or does it return paginated results?  For instance, I think we had to loop through all pages to get at all the CommunicationRequest resources when sending out email notifications (see [here](https://github.com/uwcirg/isacc-messaging-service/blob/00473047a1dd25e203b01645638d0276cf676c47/isacc_messaging/api/isacc_record_creator.py#L422)).  I am not certain if we need to consider the same, i.e. pagination, here.

**Review by achen2401 — APPROVED** [review]

**Review by pbugni — APPROVED** [review]

**Pull request merged**